src/Compilers/Core/Portable/CommandLine/EditorConfig.cs

```diff
@@ -163,6 +163,72 @@ public static EditorConfig Parse(string text, string parentDirectory)
             return new EditorConfig(globalSection, namedSectionBuilder.ToImmutable(), parentDirectory);
         }
 
+        /// <summary>
+        /// Combine an editorconfig with an editorconfig nested in a subdirectory to form a new "effective"
+        /// editorconfig. "Nested" is defined as the parent directory being an ordinal prefix of the nested
+        /// directory.
+        /// </summary>
+        /// <remarks>
+        /// Editorconfig files are combined by applying all properties from the parent editorconfig, then
+        /// applying all properties from the nested editorconfig, with any conflicts resolving in favor
+        /// of the nested editorconfig. Any new sections in the nested editorconfig will appear at the
+        /// end of <see cref="EditorConfig.NamedSections" />.
+        /// </remarks>
+        public static EditorConfig Combine(EditorConfig nested, EditorConfig parent)
+        {
+            Debug.Assert(nested.Directory.StartsWith(parent.Directory, StringComparison.Ordinal));
+
+            // PROTOTYPE(editorconfig): Global properties are not described in the editorconfig
+            // spec and are not specified as being inherited or overriden. The only mentioned
+            // global property, 'root', should definitely not be inherited. For now, let's assume
+            // that the global properties are fixed.
+            var newGlobals = nested.GlobalSection;
+
+            var newSections = ImmutableArray.CreateBuilder<Section>(
```

Comment:
>CreateBuilder [](start = 45, length = 13)

Why not pooled builder?

Reply:
This code still may end up shared with other pieces of the company, so I'm keeping away from Roslyn-specific data structures for now.